src/lambda/commands/uploadLambda.ts

```diff
@@ -179,26 +189,49 @@ class UploadLambdaWizard extends Wizard<UploadLambdaWizardState> {
             }
         })
 
-        this.form.directoryUploadType.bindPrompter(() => createDirectoryUploadPrompter(), {
-            showWhen: ({ uploadType }) => uploadType === 'directory',
+        this.form.lambda.name.bindPrompter(state => {
+            // invoking from the command palette passes no arguments
+            if (!invokePath) {
+                if (state.uploadType === 'directory') {
+                    return createFunctionNamePrompter(state.lambda!.region!, state.targetUri)
+                } else {
+                    // if a zip file is chosen, pass the parent directory
+                    return createFunctionNamePrompter(
+                        state.lambda!.region!,
+                        vscode.Uri.parse(path.dirname(state.targetUri!.fsPath))
```

Comment:
As mentioned in the tests, don't mix-and-match `Uri.parse` and the `path` module. Use `Uri.file` for parsing FS paths. Only use `Uri.parse` when you have a scheme.